#17396 Support TiDB using Modules

Icemap opened the pull request:
Hi guys. I add TiDB support for DBeaver, again.

This time I used module way, just like @serge-rider said.
[TiDB](https://docs.pingcap.com/tidb/stable/overview) is an open-source NewSQL database.
Many of our customers want to use TiDB with DBeaver.

Commit:
#17396 fork from the ob

Icemap committed:
#17936 [Debug] fix unchanged ob name, and currect the import links

Icemap committed:
#17936 [Update] add plugins to feature.xml

Icemap committed:
#17936 [Update] tidb not need to overwrite auth module

Icemap committed:
#17936 [Debug] export-package fix

Icemap committed:
#17936 [Update] support explain for tidb

Icemap committed:
#17936 [Update] support sequence start from tidb 4.0

Icemap committed:
#17936 [Update] remove fk for now

dveeden commented on plugins/org.jkiss.dbeaver.ext.tidb/plugin.xml:
```diff
@@ -0,0 +1,167 @@
+                    defaultUser="root"
+                    webURL="https://pingcap.com/"
+                    propertiesURL="https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-configuration-properties.html"
+                    description="TiDB Driver (MySQL Connector 8)"
```
```suggestion
                    description="TiDB Driver (MySQL Connector/J 8.0)"
```

dveeden commented on plugins/org.jkiss.dbeaver.ext.tidb/plugin.xml:
```diff
@@ -0,0 +1,167 @@
+                    <property name="loginTimeout" value="20000"/>
+                    <property name="connectTimeout" value="20000"/>
+
+                    <property name="enabledTLSProtocols" value="TLSv1,TLSv1.1,TLSv1.2"/>
```
```suggestion
                    <property name="enabledTLSProtocols" value="TLSv1.2,TLSv1.3"/>
```

https://datatracker.ietf.org/doc/rfc8996/

Commit:
#17396 Apply suggestions from code review

Co-authored-by: Daniël van Eeden <github@myname.nl>

Commit:
#17396 [Update] makes style check CI happy

Commit:
#17396 [Update] makes style check CI more happy

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/model/plan/TiDBPlanNodePlain.java:
```diff
@@ -0,0 +1,125 @@
+package org.jkiss.dbeaver.ext.tidb.model.plan;
```
Please add copyright

Icemap replied:
Done

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb.ui/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,24 @@
+ org.jkiss.dbeaver.ext.mysql,
+ org.jkiss.dbeaver.ext.mysql.ui,
+ org.jkiss.dbeaver.ext.tidb,
+ org.jkiss.dbeaver.ext.oracle,
```
You do not have Oracle bundle dependencies. What for it here?

Icemap replied:
I copied from here: https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.ext.oceanbase.ui/META-INF/MANIFEST.MF#L15
So deleting this line in both files is better?

LonwoLonwo replied:
Just remove it from your code.

Icemap replied:
Got it.

Commit:
#17396 [Update] add copyright to each code files

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb.ui/plugin.xml:
```diff
@@ -0,0 +1,47 @@
+                targetID="org.jkiss.dbeaver.ext.ui.newConnectionWizard"
+                label="%dialog.connection.header"
+                class="org.jkiss.dbeaver.ext.tidb.ui.views.TiDBConnectionPage"
+                icon="platform:/plugin/org.jkiss.dbeaver.ext.tidb/icons/tidb_icon.png"/>
```
So, you have `ext.tidb.ui/icons`, but you do not use them. Please remove them. `ext.tidb/icons` will be enough
Except for idb_logo.

Icemap replied:
Deleted.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb.ui/src/org/jkiss/dbeaver/ext/tidb/ui/views/TiDBConnectionPage.java:
```diff
@@ -0,0 +1,165 @@
+        DBPDataSourceContainer activeDataSource = site.getActiveDataSource();
+        DBPConnectionConfiguration connectionInfo = activeDataSource.getConnectionConfiguration();
+        if (!CommonUtils.isEmpty(connectionInfo.getHostName())) {
+            hostText.setText(CommonUtils.notEmpty(connectionInfo.getHostName()));
```
Please remove `CommonUtils.notEmpty`. You already checked it with `!CommonUtils.isEmpty`

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb.ui/src/org/jkiss/dbeaver/ext/tidb/ui/views/TiDBConnectionPage.java:
```diff
@@ -0,0 +1,165 @@
+            hostText.setText(MySQLConstants.DEFAULT_HOST);
+        }
+        if (!CommonUtils.isEmpty(connectionInfo.getHostPort())) {
+            portText.setText(CommonUtils.notEmpty(connectionInfo.getHostPort()));
```
Please remove `CommonUtils.notEmpty`. You already checked it with `!CommonUtils.isEmpty`

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb.ui/src/org/jkiss/dbeaver/ext/tidb/ui/views/TiDBConnectionPage.java:
```diff
@@ -0,0 +1,165 @@
+            portText.setText("");
+        }
+        if (!CommonUtils.isEmpty(connectionInfo.getDatabaseName())) {
+            databaseText.setText(CommonUtils.notEmpty(connectionInfo.getDatabaseName()));
```
Please remove `CommonUtils.notEmpty`. You already checked it with `!CommonUtils.isEmpty`

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,21 @@
+Bundle-ManifestVersion: 2
+Bundle-Name: %Bundle-Name
+Bundle-SymbolicName: org.jkiss.dbeaver.ext.tidb;singleton:=true
+Bundle-Version: 1.0.32.qualifier
```
Your bundle is new. So it should be 1.0.0

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb.ui/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,24 @@
+Bundle-ManifestVersion: 2
+Bundle-Name: %Bundle-Name
+Bundle-SymbolicName: org.jkiss.dbeaver.ext.tidb.ui;singleton:=true
+Bundle-Version: 1.0.32.qualifier
```
Your bundle is new. So it should be 1.0.0

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb.ui/pom.xml:
```diff
@@ -0,0 +1,16 @@
+		<relativePath>../</relativePath>
+	</parent>
+	<artifactId>org.jkiss.dbeaver.ext.tidb.ui</artifactId>
+	<version>1.0.32-SNAPSHOT</version>
```
Your bundle is new. So it should be 1.0.0

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/OSGI-INF/l10n/bundle.properties:
```diff
@@ -0,0 +1,70 @@
+tree.events.node.name=Events
+tree.events.node.tip=Events
+tree.event.node.name=Event
+tree.packages.node.name=Packages
```
Hm. But you do not read packages in your Catalog class. As procedures.

Icemap replied:
Deleted.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/OSGI-INF/l10n/bundle.properties:
```diff
@@ -0,0 +1,70 @@
+tree.plugins.node.name=Plugins
+tree.plugin.node.name=Plugin
+tree.collation.node.name=Collation
+tree.events.node.name=Events
```
Does TiDB support events?

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/OSGI-INF/l10n/bundle.properties:
```diff
@@ -0,0 +1,70 @@
+tree.references.node.name=References
+tree.reference_key.node.name=Reference Key
+tree.reference_key_columns.node.name=Reference Key columns
+tree.triggers.node.name=Triggers
```
Does TiDB support triggers?

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/model/plan/TiDBPlanNodePlain.java:
```diff
@@ -0,0 +1,125 @@
+        return id;
+    }
+
+    @Property(order = 1, name="estRows", viewable = true)
```
Please, remove `name="estRows"`
And localize this property in the osgi-inf folder.
As an example you can take a look at this code:
https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle.properties#L263
Please use a capital letter for the first letter.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/model/plan/TiDBPlanNodePlain.java:
```diff
@@ -0,0 +1,125 @@
+        return estRows;
+    }
+
+    @Property(order = 2, name="task", viewable = true)
```
Please, remove `name`
And localize this property in the osgi-inf folder.
As an example you can take a look at this code:
https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle.properties#L263
Please use a capital letter for the first letter.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/model/plan/TiDBPlanNodePlain.java:
```diff
@@ -0,0 +1,125 @@
+        return task;
+    }
+
+    @Property(order = 3, name="access object", viewable = true)
```
Please, remove `name`
And localize this property in the osgi-inf folder.
As an example you can take a look at this code:
https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle.properties#L263
Please use a capital letter for the first letter.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/model/plan/TiDBPlanNodePlain.java:
```diff
@@ -0,0 +1,125 @@
+        return accessObject;
+    }
+
+    @Property(order = 4, name="operator info", viewable = true)
```
Please, remove `name`
And localize this property in the osgi-inf folder.
As an example you can take a look at this code:
https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle.properties#L263
Please use a capital letter for the first letter.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/model/plan/TiDBPlanNodePlain.java:
```diff
@@ -0,0 +1,125 @@
+
+    @Override
+    public String getNodeType() {
+        return this.id.trim().replaceAll("└", "").replaceAll("─", "");
```
Please add any comments here in the code. Why do you replace it here?

Icemap replied:
Added.

Commit:
#17396 [Update] remove redundant CommonUtils.notEmpty

Commit:
#17396 [Update] reversion plugins to 1.0.0 for initial

Commit:
#17396 [Update] TiDBPlanNodePlain properties name using OSGI instead

Commit:
#17396 [Update] add comments for getNodeType function

Commit:
#17396 [Update] makes style check CI more more happy

Icemap committed:
#17396 [Update] follow other bundle version add .qualifier subffix

Icemap committed:
#17396 [Debug] repair irregular property get function name

Icemap committed:
#17396 [Remove] no used pictures

Commit:
#17396 [Update] add Javadoc comment and makes style check CI more *3 happy

Commit:
#17396 [Update] change Javadoc comments

Commit:
#17396 [Update] makes style check CI more *4 happy

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb.ui/OSGI-INF/l10n/bundle.properties:
```diff
@@ -0,0 +1,13 @@
+editor.schema_privileges.name=Schema Privileges
+editor.session_manager.name=Session Manager
+editor.source.name=Source
+editor.package.body.name=Body source
```
You do not have MySQLPackageBodyEditor here. So no reason for this label.

Icemap replied:
Deleted

Commit:
#17396 [Update] delete package editor osgi properties

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb.ui/src/org/jkiss/dbeaver/ext/tidb/ui/views/TiDBConnectionPage.java:
```diff
@@ -0,0 +1,165 @@
+    @Override
+    public boolean isComplete() {
+        return super.isComplete() && hostText != null && databaseText != null && portText != null
+                && !CommonUtils.isEmpty(hostText.getText());
```
Can you connect to TiDB without portText.text? You have `portText.setText("");` in your code, but I suppose you can connect without the port number.

Icemap replied:
You're right. I'll delete ` && portText != null` part.

LonwoLonwo replied:
Am. No. I think you can add CommonUtils.isNotEmpty(portText.getText())

Icemap replied:
Done

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/plugin.xml:
```diff
@@ -0,0 +1,167 @@
+    
+    <extension point="org.jkiss.dbeaver.dataTypeProvider">
+        <provider
+            class="org.jkiss.dbeaver.ext.tidb.data.TiDBValueHandlerProvider"
```
Ok, you have this class. But it is empty. What for then?

Icemap replied:
Used `org.jkiss.dbeaver.model.impl.jdbc.data.handlers.JDBCStandardValueHandlerProvider` instead.

Commit:
#17396 [Delete] no used TiDBValueHandlerProvider class

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/mysql/model/TiDBMySQLView.java:
```diff
@@ -0,0 +1,65 @@
+        List<MySQLTableColumn> childColumns = getContainer().getTiDBTableCache().getChildren(monitor,
+                getContainer(), this);
+        if (childColumns == null) {
+            return Collections.emptyList();
```
I think - this is an extraordinary situation.
I tried to do 
`create view testView as select 1`
but testView has one column in this situation.
So I think we can add logging here.
`log.error("Can't find attributes for the view " + getName())`

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/mysql/model/TiDBMySQLCatalog.java:
```diff
@@ -0,0 +1,266 @@
+                @NotNull TiDBMySQLCatalog owner, @Nullable MySQLTableBase forTable) throws SQLException {
+            if (forTable instanceof TiDBMySQLView) {
+                JDBCPreparedStatement dbStat = session
+                        .prepareStatement("desc " + owner.getName() + "." + forTable.getName());
```
Please use `forTable.getFullyQualifiedName(DBPEvaluationContext.DDL)` instead `owner.getName() + "." + forTable.getName()`
To avoid cases when a table or catalog is quoted.

LonwoLonwo commented:
Also, please add some pictures to this PR. We do not have a test TiDB environment for now. 
Connection page + Navigator with all nodes

Icemap committed:
#17396 [Update] add log and avoid quoted cases

Icemap committed:
#17396 [Update] make CI more*5 happy

Icemap commented:
> Also, please add some pictures to this PR. We do not have a test TiDB environment for now. Connection page + Navigator with all nodes

Sure.

![image](https://user-images.githubusercontent.com/26212551/203855351-83fc2ba0-3a61-4317-8c36-d7060c1cfc92.png)
![image](https://user-images.githubusercontent.com/26212551/203855381-3c1ca616-2022-4eae-84d9-3c7fe25d5e9d.png)
![image](https://user-images.githubusercontent.com/26212551/203855413-afa76d8a-7191-4bb9-bc8c-2e975fcb3246.png)
![image](https://user-images.githubusercontent.com/26212551/203855770-bfe0d96b-830c-494a-b72f-88bd384a75fd.png)

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/OSGI-INF/l10n/bundle.properties:
```diff
@@ -0,0 +1,67 @@
+tree.sequences.node.tip = TiDB Sequences
+
+meta.org.jkiss.dbeaver.ext.tidb.model.plan.TiDBPlanNodePlain.id.name = Operator ID
+meta.org.jkiss.dbeaver.ext.tidb.model.plan.TiDBPlanNodePlain.estRows.name = Astimated Rows
```
Estimated Rows?

Commit:
#17396 [Typo] Astimated -> Estimated

Icemap commented:
Hi guys, a month has passed, any progress? @serge-rider @E1izabeth

E1izabeth commented:
Waiting for @serge-rider review.

Commit:
Merge branch 'devel' into feat-tidb-module

Commit:
Merge branch 'devel' into feat-tidb-module

Commit:
Merge branch 'devel' into feat-tidb-module

dveeden commented:
@serge-rider is there anything we can do to help you review this? e.g. helping with setting up or providing a TiDB server?

LonwoLonwo commented:
Hello.

So, we found the time to discuss your PR. The main issue for now - is too much copy-paste in the code.

- Did you do TiDBConnectionPage only for the logo? Then please use inheritance from MySQLConnectionPage and override the logo/image method.

- You do not need all navigator tree in the org.jkiss.dbeaver.ext.tidb/plugin.xm - just add MySQL as a parent.

- Why did you add your custom table cache to MySQLCatalog?
- What is the point of the TiDBMySQLViewColumn and TiDBMySQLView classes?
- Do you really need to override these to caches?

```java
   private final JDBCBasicDataTypeCache<MySQLDataSource, JDBCDataType> dataTypeCache;
    private final TiDBCatalogCache tidbCatalogCache = new TiDBCatalogCache();
```

- I do not see the difference between MySQLDatabaseManager and TiDBMySQLDatabaseManager.

Commit:
Merge branch 'devel' into feat-tidb-module

Icemap commented:
Sorry for the late reply. And I finally have time to change it. I'll check and change it. Thanks for reviewing.

Icemap committed:
#17396 inheritance MySQLConnectionPage

Icemap committed:
#17396 visable fk tab for TiDB v6.6.0

Icemap committed:
#17396 visable fk tab for TiDB v6.6.0

Icemap committed:
#17396 clean the code

Icemap commented:
Hi @LonwoLonwo, thanks for the review! I already clean the code as you said. For now, it runs well and more clearly! 

> Did you do TiDBConnectionPage only for the logo? Then please use inheritance from MySQLConnectionPage and override the logo/image method.

OK, but I need to Export the `org.jkiss.dbeaver.ext.mysql.ui.views` package.

> You do not need all navigator tree in the org.jkiss.dbeaver.ext.tidb/plugin.xm - just add MySQL as a parent.

I need to customize the folder. For example, I need to check the Foreign Key tab only visible above the `6.6.0` version. And 
I also need to hide the trigger and procedure. Can I add MySQL as a parent to simplify my code? Can I get more information or documents on it?

> Why did you add your custom table cache to MySQLCatalog?
> What is the point of the TiDBMySQLViewColumn and TiDBMySQLView classes?
> Do you really need to override these to caches?
>    ```
>    private final JDBCBasicDataTypeCache<MySQLDataSource, JDBCDataType> dataTypeCache;
>    private final TiDBCatalogCache tidbCatalogCache = new TiDBCatalogCache();
>    ```
> I do not see the difference between MySQLDatabaseManager and TiDBMySQLDatabaseManager.

You're totally right! I deleted the `TiDBMySQLCatalog.java`, `TiDBMySQLViewColumn.java`, `TiDBMySQLView.java`, and `TiDBMySQLDatabaseManager.java`. Those 4 files are useless. And I adjusted the `TiDBMySQLDataSource` to clean the code.

LonwoLonwo commented:
> I need to customize the folder.

Please try to use `treeInjection` in the plugin.xml for this. But it has its own restrictions. If this does not work - tell me, and we will think about what can be done as another option.

https://github.com/dbeaver/dbeaver/blob/1fbc82abf16cb148af7dd46f47ebf5d78ed2438a/plugins/org.jkiss.dbeaver.ext.db2.i/plugin.xml#L17-L20

Icemap committed:
#17396 clean the config and write the dialect

Icemap commented:
> Please try to use `treeInjection` in the plugin.xml for this. But it has its own restrictions. If this does not work - tell me, and we will think about what can be done as another option.

Hi @LonwoLonwo, I used the `treeInjection` with `replaceChildren` param to simplify the config file. I only need to rewrite the `mysql/database` path.

And I wrote a dialect for TiDB. It's also inherited from MySQL. So I need to adjust a little bit of `MySQLDialect` and `MySQLDataSource`. Because those two classes are not designed to be the base classes.

PTAL.

Commit:
Merge branch 'devel' into feat-tidb-module

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb.ui/src/org/jkiss/dbeaver/ext/tidb/ui/internal/TiDBMessages.java:
```diff
@@ -0,0 +1,38 @@
+public class TiDBMessages extends NLS {
+    private static final String BUNDLE_NAME = "org.jkiss.dbeaver.ext.tidb.ui.internal.TiDBMessages"; //$NON-NLS-1$
+
+    public static String tidb_connection_page_label_port;
```
It looks like you do not need it anymore. This class and properties.

Icemap replied:
Deleted.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/plugin.xml:
```diff
@@ -0,0 +1,139 @@
+                    categories="sql">
+
+                    <file type="jar" path="maven:/mysql:mysql-connector-java:RELEASE" bundle="!drivers.mysql"/>
+                    <file type="license" path="drivers/mysql/LICENSE.txt" bundle="drivers.mysql"/>
```
`<file type="license" path="https://www.gnu.org/licenses/old-licenses/lgpl-2.0.txt" bundle="!drivers.mysql"/>`

Please, add the path to the license here

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/plugin.xml:
```diff
@@ -0,0 +1,139 @@
+        <manager class="org.jkiss.dbeaver.ext.tidb.mysql.model.TiDBMySQLDatabaseManager"
+            objectType="org.jkiss.dbeaver.ext.tidb.mysql.model.TiDBMySQLCatalog"
+            label="%manager.catalog.name"/>
+        <manager class="org.jkiss.dbeaver.ext.mysql.edit.MySQLForeignKeyManager" objectType="org.jkiss.dbeaver.ext.mysql.model.MySQLTableForeignKey"/>
```
```suggestion

```

You do not need it here. The parent class already has this.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/model/plan/TiDBPlanAnalyzer.java:
```diff
@@ -0,0 +1,61 @@
+        final String plainQuery = SQLUtils.stripComments(dialect, query).toUpperCase();
+        final String firstKeyword = SQLUtils.getFirstKeyword(dialect, plainQuery);
+        if (TiDBPlanAnalyzer.block(firstKeyword)) {
+            throw new DBCException("This statements could not produce execution plan");
```
```suggestion
            throw new DBCException("This statement could not produce execution plan");
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/mysql/model/TiDBDialect.java:
```diff
@@ -0,0 +1,68 @@
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource;
+import org.jkiss.utils.ArrayUtils;
+
+public class TiDBDialect extends MySQLDialect {
```
Please, add correct indentations and remove tabs from this class.

Icemap replied:
Replaced by 4 spaces.

Icemap committed:
#17396 remove the TiDBMessages class and props

Icemap committed:
Merge branch 'feat-tidb-module' of github.com:Icemap/dbeaver into feat-tidb-module

Commit:
#17396 Apply suggestions from code review

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

Icemap committed:
#17396 add license file

Icemap committed:
#17396 switch tabs to 4 spaces

Commit:
#17396 make code style checker happy

Icemap commented:
Merged and used the `checkstyle-10.8.1-all.jar` to check all of the code in 2 TiDB packages. Looks like this:

```
java -jar checkstyle-10.8.1-all.jar -c dbeaver-checkstyle-config.xml ${tidb_java_code_file_list}
Starting audit...
Audit done.
```

PTAL @LonwoLonwo .

E1izabeth commented:
Could you merge last changes from `devel` branch?

Commit:
Merge branch 'devel' into feat-tidb-module

Icemap commented:
> Could you merge last changes from `devel` branch?

Merged. @E1izabeth

Commit:
Merge branch 'devel' into feat-tidb-module

uslss commented:
- Session Manager is shown twice in DB Navigator
- Driver version proposed by default (8.0.32) cannot be downloaded due to 404 error. Works ok with 8.0.30
![image](https://user-images.githubusercontent.com/31996417/226357420-7e8bdee0-e34f-4ae2-ad20-7f19731102e1.png)

uslss commented:
- Incorrect ddl is generated for sequences
![image](https://user-images.githubusercontent.com/31996417/226357889-278ad961-0a44-449b-818f-2eeb74f223b9.png)

dveeden commented:
> * Session Manager is shown twice in DB Navigator
> 
>     * Driver version proposed by default (8.0.32) cannot be downloaded due to 404 error. Works ok with 8.0.30
>       ![image](https://user-images.githubusercontent.com/31996417/226357420-7e8bdee0-e34f-4ae2-ad20-7f19731102e1.png)

- Old URL: https://repo1.maven.org/maven2/mysql/mysql-connector-java/8.0.30/mysql-connector-java-8.0.30.jar
- New URL: https://repo1.maven.org/maven2/com/mysql/mysql-connector-j/8.0.32/mysql-connector-j-8.0.32.jar

https://repo1.maven.org/maven2/mysql/mysql-connector-java/8.0.32/mysql-connector-java-8.0.32.pom has this:
```
<relocation>
<groupId>com.mysql</groupId>
<artifactId>mysql-connector-j</artifactId>
<message>
MySQL Connector/J artifacts moved to reverse-DNS compliant Maven 2+ coordinates.
</message>
</relocation>
```

Icemap commented:
Thanks for testing, @uslss. I'll fix it later.

Commit:
Merge branch 'devel' into feat-tidb-module

Icemap committed:
#17396 fix mysql-connector-j download url & session manager shown twice

Icemap committed:
Merge branch 'feat-tidb-module' of github.com:Icemap/dbeaver into feat-tidb-module

Icemap commented:
Hi, @uslss. I already fixed those two bugs:

1. Session Manager is shown twice in DB Navigator
    - Fix function: Deleted the redundant Session Manager config.
2. Driver version proposed by default (8.0.32) cannot be downloaded due to 404 error. Works ok with 8.0.30
    - Fix function: Replaced the new MySQL driver URL. Thanks for @dveeden.

But this:

> Incorrect ddl is generated for sequences

I'm using the `org.jkiss.dbeaver.ext.mysql.model.MySQLSequence` class to offer this feature. So these two ddl unequal situations also appear on the MariaDB. I think this is the feature, not a bug.
<img width="1173" alt="image" src="https://user-images.githubusercontent.com/26212551/226506480-7541018e-9b19-4b0a-ae3c-a1b303e52037.png">

uslss commented:
verified
